src/server/pps/server/api_server.go

```diff
@@ -2603,6 +2603,15 @@ func (a *apiServer) RunPipeline(ctx context.Context, request *pps.RunPipelineReq
 	key := path.Join
 	requestProv := make(map[string]*pfs.CommitProvenance)
 	for _, prov := range request.Provenance {
+		if prov == nil {
+			return nil, fmt.Errorf("request should not contain nil provenance")
+		}
+		if prov.Branch == nil {
+			return nil, fmt.Errorf("request provenance cannot have a nil branch")
+		}
+		if prov.Branch.Repo == nil {
+			return nil, fmt.Errorf("request provenance branch must have a non nil repo")
+		}
```

Comment:
What did the request in the initial issue look like? I'm wondering if the cli command maybe has a bug in it too, since I wouldn't expect it to be able to send inputs like this. These changes are definitely good and should be merged, I'm just wondering if a few fixes in the command could be worthwhile too here.